File: Data Collection/@types/media.ts
export interface article{ // Article object
    outletName: string
    headline:string,
    description: string,
    author: string,
    publishDate: Date,
    isLegacy: boolean,
    sentimentScore: number,
    // Modern article data points below, legacy and modern data points above
    linkToArticle?: string,
    imageURL?: string,
    catergories?: string[]
}

export interface mediaOutlet{
    name: string,
    rssLink:string,
    articleList: article[]
}

export interface rssReply{
    title:string,
    description: string,
    creator: string,
    isoDate: string,
    isLegacy: boolean,
    link: string,
    content: string,
    contentSnippet: string,
    categories: string[]
}

File: Tooling/src/media.py
class Outlet:
    def __init__(self, name):
        self.name = name
        self.articleList = []

    def addArticle(self, article) -> None:
        self.articleList.append(article)

class Article:
    def __init__(self, outlet, headline, description, author, date, sentimentScore, url, catergories):
        self.outlet = outlet
        self.headline = headline
        self.description = description
        self.author = author
        self.date = date
        self.sentimentScore = sentimentScore
        self.url = url
        self.catergories = catergories